### ISSUE TITLE
XSS protection disabling is broken

### ISSUE THREAD

**Issue by marcingajda** [bolt/bolt#7442]
When you have a field with HTML and you open the preview page Bolt sends that code in POST parameters. Chrome has a feature that detects HTML code in POST and blocks displaying the response to protect from XSS. It can be disabled however by the server by sending a header `X-XSS-Protection` set `0`. Bolt tries to use that trick but fails with it. 

Details
-------

 - Relevant Bolt Version: 3.4.8 and probably later
 - Install type: Composer install
 - PHP version: 7.1
 - Used web server: Nginx version 1.13.9
 - Browser: Chrome 65.0.3325.181

Reproduction
------------

### Bug summary

Preview button doesn't work on Chrome if you have an field with HTML in the content type editor. It can be a field for embeding an iframe or just a Tweet added with the build-in embeding tool.

### Specifics

The [`DisableXssProtectionListener`](https://github.com/bolt/bolt/blob/14b3ec1019fd3a6af91ff92e246ec9081e50e49f/src/EventListener/DisableXssProtectionListener.php) which adds that header is runed when the current route name is in the array defined in the [`disable_xss_protection_routes`](https://github.com/bolt/bolt/blob/180a3a4c21e37fe55a7c681657f53e1ada682c81/src/Provider/EventListenerServiceProvider.php#L30). The problem is when you click the Preview button Bolt is using `contentlink` route, not the `preview` one. I checked it by inserting `dump($route)` in the mentioned listener below 44 line.

### Steps to reproduce

* Add to any content type a text field
* Go to record editor
* Put in that field `<iframe src="http://google.com"></iframe>`
* Click the preview button

### Expected result

Preview of that record should be displayed

### Actual result

![image](https://user-images.githubusercontent.com/6128754/38773669-cbf18c98-4052-11e8-900e-45cda3b65b5d.png)

How to fix it (temporarly)?
-------------

If you have a bundled extension you can overwrite there the list of not protected routes
```
$app['disable_xss_protection_routes'] = [
    'preview',
    'fileedit',
    'contentlink' // <= that helps
];
```
However this is probably a workaround and just the route used for previews is wrong. That is also why I didn't create a PR.

**Comment by bobdenotter** [comment]
Pinging @JarJak, @marcingajda. Hi, any progress on this one, yet? ;-)

**Comment by marcingajda** [comment]
@bobdenotter I will prepare a PR in free time ;)

**Comment by GwendolenLynch** [comment]
Closed by #7458
